cg/models/demultiplex/run_parameters.py

```diff
@@ -43,82 +46,94 @@ def reagent_kit_version(self) -> str:
+        message = f"Unknown flow cell type {xml_node.text}"
         LOG.warning(message)
         raise FlowCellError(message)
 
+    @property
+    def flow_cell_mode(self) -> Optional[str]:
+        """Return the flow cell mode."""
+        node_name = "/RfidsInfo/FlowCellMode"
+        xml_node = self.tree.find(node_name)
+        if xml_node is None:
+            LOG.warning("Could not determine flow cell mode")
+            LOG.info("Set flow cell mode to None")
+            return
+        return xml_node.text
+
     @property
     def run_type(self) -> Literal["wgs", "fluffy"]:
-        """Fetch what type of run the parameters is"""
+        """Return the sequencing run type."""
         if self.index_length == 8:
             return "fluffy"
         return "wgs"
 
     @staticmethod
     def node_not_found(node: Optional[ElementTree.Element], name: str) -> None:
-        """Raise exception if node if not found"""
+        """Raise exception if the given node is not found."""
         if node is None:
             message = f"Could not determine {name}"
             LOG.warning(message)
             raise FlowCellError(message)
 
     def get_node_integer_value(self, node_name: str, name: str) -> int:
+        """Return the value of the node as an integer."""
         xml_node = self.tree.find(node_name)
         self.node_not_found(node=xml_node, name=name)
         return int(xml_node.text)
 
-    def index_read_one(self) -> int:
-        """Get the value for index read one"""
+    def get_index1_cycles(self) -> int:
+        """Return the number of cycles in the first index read."""
         node_name = "./IndexRead1NumberOfCycles"
         return self.get_node_integer_value(node_name=node_name, name="length of index one")
 
-    def index_read_two(self) -> int:
-        """Get the value for index read one"""
+    def get_index2_cycles(self) -> int:
+        """Return the number of cycles in the second index read."""
         node_name = "./IndexRead2NumberOfCycles"
         return self.get_node_integer_value(node_name=node_name, name="length of index two")
 
-    def read_one_nr_cycles(self) -> int:
-        """Get the nr of cycles for read one"""
+    def get_read1_cycles(self) -> int:
+        """Return the number of cycles in the first read."""
         node_name = "./Read1NumberOfCycles"
         return self.get_node_integer_value(node_name=node_name, name="length of reads one")
 
-    def read_two_nr_cycles(self) -> int:
-        """Get the nr of cycles for read one"""
+    def get_read2_cycles(self) -> int:
+        """Return the number of cycles in the second read."""
         node_name = "./Read2NumberOfCycles"
         return self.get_node_integer_value(node_name=node_name, name="length of reads two")
 
-    def base_mask(self) -> str:
-        """create the bcl2fastq basemask for novaseq flowcells
+    def get_base_mask(self) -> str:
+        """Create the bcl2fastq basemask for novaseq flow cells.
```

Comment:
```suggestion
        """Create the basemask for novaseq flow cells.
```
Also for Dragen?

Reply:
In theory yes, I don't know why it says only bcl2fastq. Currently, there is no usage for this in cg. I don't know why it was implemented, but I know we will have to use it later to override the cycles in (I believe) both dragen and bcl2fastq.